applications/StructuralMechanicsApplication/custom_elements/base_solid_element.h

```diff
@@ -50,6 +50,68 @@ namespace Kratos
 class KRATOS_API(STRUCTURAL_MECHANICS_APPLICATION)  BaseSolidElement
     : public Element
 {
+protected:
+    /**
+     * Internal variables used in the kinematic calculations
+     */
+    struct KinematicVariables
+    {
+        Vector  StrainVector;
+        Vector  StressVector;
+        Vector  N;
+        Matrix  B;
```

Comment:
B is different to each GP, so it is supposed to be defined here or in a similar structure